yo1k/tic_tac_toe/test/core/test_game.py

```diff
@@ -0,0 +1,264 @@
+        self.assertEqual(game_state.players[0].wins, 1)
+        self.assertEqual(game_state.players[1].wins, 0)
+
+    def test_advance_draw_action(self):
+        previous_game_state_board = [
+            [g.Mark.O, g.Mark.X, g.Mark.O],
+            [g.Mark.O, g.Mark.X, g.Mark.X],
+            [g.Mark.X, g.Mark.O, None]]
+        game_state = StateSetup._set_state(
+            board=g.Board(previous_game_state_board),
+            step=8)
+        act_queue_p1 = ListActionQueue([g.Action.new_occupy(g.Cell(2, 2))])
+        g.Logic([act_queue_p1, []]).advance(game_state)
+        next_game_state_board = [
+            [g.Mark.O, g.Mark.X, g.Mark.O],
+            [g.Mark.O, g.Mark.X, g.Mark.X],
+            [g.Mark.X, g.Mark.O, g.Mark.X]]
+        self.assertEqual(game_state.board.cells, next_game_state_board)
+        self.assertEqual(game_state.step, 8)
+        self.assertEqual(game_state.round, 0)
+        self.assertEqual(game_state.phase, g.Phase.OUTROUND)
+        self.assertEqual(game_state.players[0].wins, 0)
+        self.assertEqual(game_state.players[1].wins, 0)
+
+    def test_advance_next_round_action_outround(self):
+        previous_game_state_board = [
+            [g.Mark.X, g.Mark.O, g.Mark.X],
+            [None, g.Mark.X, g.Mark.O],
+            [g.Mark.X, None, g.Mark.O]]
+        game_state = StateSetup._set_state(
+            player1=StateSetup._set_player(g.Mark.X, 1),
+            board=g.Board(previous_game_state_board),
+            phase=g.Phase.OUTROUND, step=6)
+        act_queue_p1 = ListActionQueue([g.Action.new_start()])
+        g.Logic([act_queue_p1, []]).advance(game_state)
+        next_game_state_board = [[None, None, None], [None, None, None], [None, None, None]]
+        self.assertEqual(game_state.board.cells, next_game_state_board)
+        self.assertEqual(game_state.step, 0)
+        self.assertEqual(game_state.round, 1)
+        self.assertEqual(game_state.phase, g.Phase.INROUND)
+        self.assertEqual(game_state.players[0].wins, 1)
+        self.assertEqual(game_state.players[1].wins, 0)
+
+    def test_advance_next_round_action_onset(self):
+        game_state = StateSetup._set_state(phase=g.Phase.ONSET)
+        act_queue_p1 = ListActionQueue([g.Action.new_start()])
+        g.Logic([act_queue_p1, []]).advance(game_state)
+        next_game_state_board = [[None, None, None], [None, None, None], [None, None, None]]
+        self.assertEqual(game_state.board.cells, next_game_state_board)
+        self.assertEqual(game_state.step, 0)
+        self.assertEqual(game_state.round, 0)
+        self.assertEqual(game_state.phase, g.Phase.INROUND)
+        self.assertEqual(game_state.players[0].wins, 0)
+        self.assertEqual(game_state.players[1].wins, 0)
+
+
+class LogicQueueActionTest(unittest.TestCase):
+    def test_win_cur_round_0(self):
+        game_state = StateSetup._set_state(phase=g.Phase.ONSET)
+        act_queue_p1 = ListActionQueue([
```

Comment:
Done.
 I have used lowercase for "x" and "o" because according to [PEP8](https://www.python.org/dev/peps/pep-0008/#:~:text=Function%20names%20should,retain%20backwards%20compatibility.) "mixedCase is allowed only in contexts where that's already the prevailing style (e.g. threading.py), to retain backwards compatibility".